cpp/include/Ice/IconvStringConverter.h

```diff
@@ -126,30 +166,6 @@ IconvStringConverter<charT>::IconvStringConverter(const std::string& internalCod
     {
         throw Ice::IconvInitializationException(__FILE__, __LINE__, sce.reason());
     }
-
```

Comment:
I would replace the close(createDescriptors()) above by a allocate/deallocate of your new DescriptorHolder object (on the stack).

This way, you can move the close code to the destructor of DescriptorHolder, and make createDescriptors a private static function of DescriptorHolder. And you can probably make DescriptorHolder::descriptors const.